### PR TITLE
GenAI: update SNS topic + add browser alarm

### PR DESCRIPTION
**Pull request by sanchitmalhotra126**
Updates the SNS topic for the GenAI alarms to the "Student Learning Workday Urgent" topic, and adds a new alarm that tracks browser failure rates (now that we can replace SEARCH queries with https://github.com/code-dot-org/code-dot-org/pull/62095). Note that the SNS topic is not yet hooked into to any integrations; this may need to be manually configured in the console, so will look into that as a separate task.

## Links

https://codedotorg.atlassian.net/browse/LABS-1170

## Testing story

Confirmed that alarms appear in Cloudwatch: https://us-east-1.console.aws.amazon.com/cloudwatch/home?region=us-east-1#alarmsV2:alarm/genai_chat_completion_high_browser_failure_rate

## Follow-up work

Add Slack integration to SNS topic.

### CODE REVIEW

File: aws/cloudformation/standalone/gen_ai_curriculum/monitoring/alarms/alarmConfigurations.ts [hunk 1]
```diff
@@ -65,38 +69,103 @@ const failureMetrics = modelIds.map((modelId, index) => ({
...
+        ReturnData: false,
+        Expression: 'SUM([m1, m2, m3, m4, m5])',
+      },
+      ...startMetrics,
+    ],
+  };
+
+const browserIndices = BROWSERS.map((_, i) => i + 1);
+
+// TODO: Unify this code with dashboard metrics if possible
```
Comment (sanchitmalhotra126) [inline aws/cloudformation/standalone/gen_ai_curriculum/monitoring/alarms/alarmConfigurations.ts:112]: This alarm more or less uses the same metrics as the equivalent dashboard widget, but alarms and dashboard syntax are unfortunately a bit different so it's a bit tricky to reuse code cleanly. I'm sure we can factor stuff out and organize this in a more modular fashion, but I figured that's something we can do as a follow-up.

File: aws/cloudformation/standalone/gen_ai_curriculum/monitoring/alarms/alarmConfigurations.ts [hunk 2]
```diff
@@ -1,8 +1,11 @@
-import {ComparisonOperator} from '@aws-sdk/client-cloudwatch';
+import {
+  ComparisonOperator,
+  PutMetricAlarmInput,
```
Comment (fisher-alice) [inline aws/cloudformation/standalone/gen_ai_curriculum/monitoring/alarms/alarmConfigurations.ts:3]: Nice.

File: aws/cloudformation/standalone/gen_ai_curriculum/monitoring/alarms/alarmConfigurations.ts [hunk 3]
```diff
@@ -65,38 +69,103 @@ const failureMetrics = modelIds.map((modelId, index) => ({
...
+    OKActions: [],
+    AlarmActions: [SNS_TOPIC],
+    InsufficientDataActions: [],
+    EvaluationPeriods: 5,
+    DatapointsToAlarm: 5,
+    Threshold: 10,
+    ComparisonOperator: ComparisonOperator.GreaterThanThreshold,
+    TreatMissingData: 'missing',
+    Metrics: [
+      ...['ChatCompletionRequestInitiated', 'ChatCompletionErrorUnhandled']
```
Comment (fisher-alice) [inline aws/cloudformation/standalone/gen_ai_curriculum/monitoring/alarms/alarmConfigurations.ts:128]: Curious what you think about setting up an alarm for when the rate limiting is higher than expected.

**Review by fisher-alice — APPROVED** [review]
Just left a minor question - looks great!
Nit: if you have time, could include https://github.com/code-dot-org/code-dot-org/pull/62095/files#r1821083625. Thanks!

File: aws/cloudformation/standalone/gen_ai_curriculum/monitoring/alarms/alarmConfigurations.ts (hunk 3 above)
Comment (sanchitmalhotra126) [reply aws/cloudformation/standalone/gen_ai_curriculum/monitoring/alarms/alarmConfigurations.ts:128]: Good question, something we could probably consider, but seems lower urgency? Maybe a good fit for the "LowUrgency" SNS topic when we set that up

**Comment by sanchitmalhotra126** [comment]
> Nit: if you have time, could include https://github.com/code-dot-org/code-dot-org/pull/62095/files#r1821083625. Thanks!

Thanks for the reminder! Logging a separate ticket to keep this change isolated https://codedotorg.atlassian.net/browse/LABS-1289

File: aws/cloudformation/standalone/gen_ai_curriculum/monitoring/alarms/alarmConfigurations.ts (hunk 3 above)
Comment (fisher-alice) [reply aws/cloudformation/standalone/gen_ai_curriculum/monitoring/alarms/alarmConfigurations.ts:128]: Nice idea!

**Pull request merged**